src/Corax/IndexEntryBuilder.cs

```diff
@@ -473,51 +504,85 @@ internal IndexEntryFieldIterator(IndexEntryFieldType type)
                 Unsafe.SkipInit(out _spanTableOffset);
+                Unsafe.SkipInit(out _nullTableOffset);
                 Unsafe.SkipInit(out _spanOffset);
                 Unsafe.SkipInit(out _longOffset);
                 Unsafe.SkipInit(out _doubleOffset);
                 Unsafe.SkipInit(out Count);
                 return;
             }
 
-            Count = VariableSizeEncoding.Read<ushort>(_buffer, out length, offset);
-
+            Count = VariableSizeEncoding.Read<ushort>(_buffer, out var length, offset);
             offset += length;
 
-            _spanTableOffset = MemoryMarshal.Read<int>(_buffer[offset..]);
-            if (((byte)Type & (byte)IndexEntryFieldType.Tuple) != 0)
+            _nullTableOffset = MemoryMarshal.Read<int>(_buffer[offset..]);
+            if (Type.HasFlag(IndexEntryFieldType.Tuple))
             {
-                _longOffset = MemoryMarshal.Read<int>(_buffer[(offset+sizeof(int))..]);
-                _doubleOffset = (offset + 2 * sizeof(int));
-                _spanOffset = (_doubleOffset + Count * sizeof(double));
+                _longOffset = MemoryMarshal.Read<int>(_buffer[(offset + sizeof(int))..]);
+                _doubleOffset = (offset + 2 * sizeof(int)); // Skip the pointer from sequences and longs.
+
+                if (Type.HasFlag(IndexEntryFieldType.HasNull))
+                {
+                    int nullBitStreamSize = Count / (sizeof(long) * 8) + Count % (sizeof(long) * 8) == 0 ? 0 : 1;
+                    _spanTableOffset = _nullTableOffset + nullBitStreamSize; // Point after the null table.                             
+                }
+                else
+                {
+                    _spanTableOffset = _nullTableOffset;
+                }
+
+                _spanOffset = (_doubleOffset + Count * sizeof(double)); // Skip over the doubles array, and now we are sitting at the start of the sequences table.
             }
             else
             {
-                _doubleOffset = 0;
-                _spanOffset = (offset + sizeof(int));
+                _doubleOffset = 0;                
                 _longOffset = 0;
+
+                if (Type.HasFlag(IndexEntryFieldType.HasNull))
+                {
+                    int nullBitStreamSize = Count / (sizeof(long) * 8) + Count % (sizeof(long) * 8) == 0 ? 0 : 1;
+                    _spanTableOffset = _nullTableOffset + nullBitStreamSize; // Point after the null table.         
+                }
+                else
+                {
+                    _spanTableOffset = _nullTableOffset;
+                }
+
+                _spanOffset = (offset + sizeof(int));
             }
 
             _currentIdx = -1;
             IsValid = true;
         }
 
+        public bool IsNull
+        {
+            get
+            {
+                if (!Type.HasFlag(IndexEntryFieldType.HasNull))
+                    return false;
+
+                throw new NotImplementedException();
```

Comment:
??

Reply:
Still not implemented... I use that to be able to kill the cases where I use that feature :)